packages/siopv2-oid4vp-op-auth/src/agent/DidAuthSiopOpAuthenticator.ts

```diff
@@ -382,21 +425,40 @@ export class DidAuthSiopOpAuthenticator implements IAgentPlugin {
       context,
     )
 
-    const contentType = response.headers.get('content-type') || ''
+    const contentType = response?.headers.get('content-type') || ''
     let responseBody: any = null
 
-    const text = await response.text()
+    const text = await response?.text()
     if (text) {
       responseBody = contentType.includes('application/json') || text.startsWith('{') ? JSON.parse(text) : text
     }
 
     return {
       body: responseBody,
-      url: response.url,
-      queryParams: decodeUriAsJson(response.url),
+      url: response?.url,
+      queryParams: response && decodeUriAsJson(response?.url),
     }
   }
 
+  private hasMDocCredentials = (credentials: UniqueDigitalCredential[]): boolean => {
+    return credentials.some(credential =>
+          credential.digitalCredential.documentFormat === CredentialDocumentFormat.MSO_MDOC &&
+          credential.digitalCredential.documentType === DocumentType.VC
+    );
+  };
+
+  private hasSdJwtCredentials = (credentials: UniqueDigitalCredential[]): boolean => {
+    return credentials.some(credential =>
+      credential.digitalCredential.documentFormat === CredentialDocumentFormat.SD_JWT &&
+      credential.digitalCredential.documentType === DocumentType.VC
+    );
+  };
+
+  private retrieveEncodedCredential = (credential: UniqueDigitalCredential) => {
+    // FIXME Remove any
```

Comment:
Why change to any to begin with. Check of original is present or not

Reply:
Removed "any"